File: Ahorn/entities/Bunny.jl
module Sardine7Bunny

using ..Ahorn, Maple

@mapdef Entity "Sardine7/Bunny" Bunny(x::Integer, y::Integer)

const placements = Ahorn.PlacementDict(
    "Bunny (Sardine7)" => Ahorn.EntityPlacement(
        Bunny
    )
)

const sprite = "scenery/Sardine7/bunny/idle00.png"

function Ahorn.selection(entity::Bunny)
    x, y = Ahorn.position(entity)

    return Ahorn.getSpriteRectangle(sprite, x, y, jx=0.5, jy=1.0)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::Bunny, room::Maple.Room)
    Ahorn.drawSprite(ctx, sprite, 0, 0, jx=0.5, jy=1.0)
end

end

File: Loenn/entities/bunny.lua
local drawableSprite = require("structs.drawable_sprite")

local bunny = {}

bunny.name = "Sardine7/Bunny"
bunny.depth = -9999
bunny.placements = {
    name = "bunny"
}

local offsetY = -4
local texture = "scenery/Sardine7/bunny/idle00"

function bunny.sprite(room, entity)
    local sprite = drawableSprite.fromTexture(texture, entity)

    sprite.y += offsetY

    return sprite
end

return bunny
